### PR TITLE
[14.0] delivery_send_to_shipper_at_operation: keep and split tracking with a comma

### PR DESCRIPTION
**Pull request by ne1800**
Keeps and splits the tracking in multi pickings.

### CODE REVIEW

**Review by jbaudoux — APPROVED** [review]
Thanks.
Can you squash the commits and rewrite the commit message as "delivery_send_to_shipper_at_operation: fix shipping tracking ref"

File: delivery_send_to_shipper_at_operation/tests/test_send_to_shipper.py [hunk 1]
```diff
@@ -150,6 +150,40 @@ def test_send_to_shipper_on_pack(self):
...
+            # Check multi tracking
+            self.assertEqual(
+                self.shipping.carrier_tracking_ref,
+                carrier_tracking_ref + "," + self.packing.carrier_tracking_ref,
+            )
+
+            self._validate_picking(self.shipping)
+            self.assertEqual(self.shipping.state, "done")
+            self.assertTrue(self.shipping.delivery_notification_sent)
+            self.assertIn(self.delivery_fee, self.order.order_line.product_id)
```
Comment (sebalix) [inline delivery_send_to_shipper_at_operation/tests/test_send_to_shipper.py:185]: Maybe not necessary to recheck every steps as it didn't change from the previous test (I only removed some `assert` statements):
```suggestion
        with mock.patch.object(
            type(self.carrier_on_pack),
            "send_shipping",
            return_value=SEND_SHIPPING_RETURN_VALUE,
        ):
            carrier_tracking_ref = "99999"
            self.shipping.carrier_id = self.carrier_on_pack
            self.shipping.carrier_tracking_ref = carrier_tracking_ref
            self._validate_picking(self.picking)
            self._validate_picking(self.packing)
            self.assertTrue(self.shipping.delivery_notification_sent)

            # Check multi tracking
            self.assertEqual(
                self.shipping.carrier_tracking_ref,
                carrier_tracking_ref + "," + self.packing.carrier_tracking_ref,
            )

            self._validate_picking(self.shipping)
            self.assertEqual(self.shipping.state, "done")
            self.assertTrue(self.shipping.delivery_notification_sent)
            self.assertIn(self.delivery_fee, self.order.order_line.product_id)
```

File: delivery_send_to_shipper_at_operation/models/stock_picking.py [hunk 2]
```diff
@@ -53,7 +53,10 @@ def _handle_send_to_shipper_at_operation(self):
                 self.delivery_notification_sent = True
                 related_ship.delivery_notification_sent = True
                 related_ship.carrier_price = self.carrier_price
-                related_ship.carrier_tracking_ref = self.carrier_tracking_ref
+                if not related_ship.carrier_tracking_ref:
+                    related_ship.carrier_tracking_ref = self.carrier_tracking_ref
+                else:
+                    related_ship.carrier_tracking_ref += "," + self.carrier_tracking_ref
```
Comment (sebalix) [inline delivery_send_to_shipper_at_operation/models/stock_picking.py:59]: I would start with the positive check, easier to read:
```suggestion
                if related_ship.carrier_tracking_ref:
                    related_ship.carrier_tracking_ref += "," + self.carrier_tracking_ref
                else:
                    related_ship.carrier_tracking_ref = self.carrier_tracking_ref
```

**Review by sebalix — APPROVED** [review]
LG, only minor remarks

**Pull request merged**